src/Place/Events/PlaceFromUDB2.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Place\Events;
+
+use CultureFeed_Cdb_Data_ActorDetail;
+use CultureFeed_Cdb_Data_Address;
+use CultureFeed_Cdb_Item_Actor;
+use CultuurNet\UDB3\Address\Address;
+use CultuurNet\UDB3\Address\Locality;
+use CultuurNet\UDB3\Address\PostalCode;
+use CultuurNet\UDB3\Address\Street;
+use CultuurNet\UDB3\Language;
+use CultuurNet\UDB3\Model\ValueObject\Geography\CountryCode;
+use CultuurNet\UDB3\Title;
+
+trait PlaceFromUDB2
+{
+    public function toGranularEvents(): array
+    {
+        $granularEvents = [];
+        $cultureFeedActor = $this->getCultureFeedActor();
+        $details = $cultureFeedActor->getDetails();
+        $firstDetail = $details->getFirst();
+
+        $granularEvents[] = new TitleUpdated($this->actorId, new Title($firstDetail->getTitle()));
+
+        $details->next();
+        while ($details->valid()) {
+            /** @var CultureFeed_Cdb_Data_ActorDetail $detail */
+            $detail = $details->current();
+            $granularEvents[] = new TitleTranslated(
+                $this->actorId,
+                new Language($detail->getLanguage()),
+                new Title($detail->getTitle())
+            );
+            $details->next();
+        }
+        /** @var CultureFeed_Cdb_Data_Address $address */
+        $address = $cultureFeedActor->getContactInfo()->getAddresses()[0];
+        $physicalAddress = $address->getPhysicalAddress();
+
+        $granularEvents[] = new AddressUpdated(
+            $this->actorId,
+            new Address(
+                new Street($physicalAddress->getStreet() . ' ' . $physicalAddress->getHouseNumber()),
+                new PostalCode($physicalAddress->getZip()),
+                new Locality($physicalAddress->getCity()),
+                new CountryCode($physicalAddress->getCountry())
+            )
+        );
+
+        return $granularEvents;
+    }
+
+    private function getCultureFeedActor(): CultureFeed_Cdb_Item_Actor
```

Comment:
Unfortunately, Conversion from a complex `XML` with namespaces & attributes to an array is not as simple as `Json::decodeAssociatively(Json::encode($xml))`

Reply:
And there is no feasible way to parse the XML using `SimpleXMLElement` without CultureFeed? Because with this implementation, we won't be able to remove the `cultuurnet/culturefeed-php` dependency in the future.

Reply:
I've found several gists to convert XML with namespaces and attributes to JSON, and they also worked.
On the one hand it would mean introducing new extra code to `udb3-backend`, but it would be a lot less than `cultuurnet/culturefeed-php`. So in the long run it might be a good thing?